keras_cv/layers/preprocessing/resizing.py

```diff
@@ -0,0 +1,287 @@
+      interpolation: String, the interpolation method. Defaults to `"bilinear"`.
+        Supports `"bilinear"`, `"nearest"`, `"bicubic"`, `"area"`, `"lanczos3"`,
+        `"lanczos5"`, `"gaussian"`, `"mitchellcubic"`.
+      crop_to_aspect_ratio: If True, resize the images without aspect
+        ratio distortion. When the original aspect ratio differs from the target
+        aspect ratio, the output image will be cropped so as to return the
+        largest possible window in the image (of size `(height, width)`) that
+        matches the target aspect ratio. By default
+        (`crop_to_aspect_ratio=False`), aspect ratio may not be preserved.
+    """
+
+    def __init__(
+        self,
+        height,
+        width,
+        interpolation="bilinear",
+        crop_to_aspect_ratio=False,
+        pad_to_aspect_ratio=False,
+        bounding_box_format=None,
+        **kwargs,
+    ):
+        self.height = height
+        self.width = width
+        self.interpolation = interpolation
+        self.crop_to_aspect_ratio = crop_to_aspect_ratio
+        self.pad_to_aspect_ratio = pad_to_aspect_ratio
+        self._interpolation_method = keras_cv.utils.get_interpolation(interpolation)
+        self.bounding_box_format = bounding_box_format
+
+        if pad_to_aspect_ratio and crop_to_aspect_ratio:
+            raise ValueError(
+                "`Resizing()` expects either `crop_to_aspect_ratio` or "
+                "`pad_to_aspect_ratio`, but not both."
+            )
+        if crop_to_aspect_ratio and bounding_box_format:
+            # TODO(lukewood): support `bounding_box.smart_resize()`
+            raise ValueError(
+                "Resizing() does not support `crop_to_aspect_ratio=True` "
+                "and `bounding_box_format` at the same time.  In order to resize with "
+                "bounding boxes, please pass `crop_to_aspect_ratio=False`."
+            )
+        super().__init__(**kwargs)
+
+    def _augment(self, inputs):
+        images = inputs.get("images", None)
+        bounding_boxes = inputs.get("bounding_boxes", None)
+
+        if images is not None:
+            images = tf.expand_dims(images, axis=0)
+            inputs["images"] = images
+
+        if bounding_boxes is not None:
+            bounding_boxes = tf.expand_dims(bounding_boxes, axis=0)
+            inputs["bounding_boxes"] = bounding_boxes
+
+        outputs = self._batch_augment(inputs)
+
+        if images is not None:
+            images = tf.squeeze(outputs["images"], axis=0)
+            inputs["images"] = images
+
+        if bounding_boxes is not None:
+            bounding_boxes = tf.squeeze(outputs["bounding_boxes"], axis=0)
+            inputs["bounding_boxes"] = bounding_boxes
+        return inputs
+
+    def _resize_with_distortion(self, inputs):
+        images = inputs.get("images", None)
+        bounding_boxes = inputs.get("bounding_boxes", None)
+        if bounding_boxes is not None:
```

Comment:
Seems to me like this guard clause should live in `init` with the one about `crop_to_aspect_ratio`

Reply:
Good point; the guard class in batch augment will catch the case where no format is specified too so we have full coverage.